akka-typed-tests/src/test/java/akka/typed/javadsl/BehaviorBuilderTest.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.typed.javadsl;
+
+import akka.typed.Behavior;
+import akka.typed.Terminated;
+import org.junit.Test;
+import org.scalatest.junit.JUnitSuite;
+
+import java.util.ArrayList;
+
+import static akka.typed.javadsl.Actor.same;
+import static akka.typed.javadsl.Actor.stopped;
+
+public class BehaviorBuilderTest extends JUnitSuite {
+    interface Message {
+    }
+
+    static final class One implements Message {
+        public String foo() {
+          return "Bar";
+        }
+    }
+    static final class MyList<T> extends ArrayList<T> implements Message {
+    };
+
+    @Test
+    public void shouldCompile() {
+      Behavior<Message> b = Actor.immutable(Message.class)
```

Comment:
looking at this and the [higher-order syntax](https://github.com/akka/akka/pull/22834) I think this is probably easier to understand